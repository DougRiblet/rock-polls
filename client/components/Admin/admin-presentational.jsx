// @flow

import React from 'react';
import { Link } from 'react-router-dom';

type PollOption = {
  question: string,
  date: string,
};

type PollOptions = {
  [key: string]: PollOption
};

type Props = {
  grabMyPolls: () => mixed,
  myPolls: PollOptions,
  user_id: string,
  username: string,
};

export default class Admin extends React.Component<Props> {
  constructor(props: Props) {
    super(props);
    this.showLoading = this.showLoading.bind(this);
    this.showListing = this.showListing.bind(this);
  }

  componentDidMount() {
    this.props.grabMyPolls(this.props.user_id);
  }

  showLoading() {
    return <p className='loading'>L O A D I N G</p>;
  }

  showListing(myPolls: PollOptions) {
    const mp = Object.entries(myPolls);
    return mp
      .sort((a, b) => new Date(b[1].date) - new Date(a[1].date))
      .map(p => (
        <li key={p[0]}>
          <Link to={`/admin/${p[0]}`}>{p[1].question}</Link>
        </li>
      ));
  }

  render() {
    if (!this.props.myPolls || !Object.keys(this.props.myPolls).length) {
      return (
        <div id='show-my-polls'>
          { this.showLoading() }
        </div>
      );
    }

    return (
      <div id='show-my-polls'>
        <ul>
          { this.showListing(this.props.myPolls) }
        </ul>
      </div>
    );
  }
}

//  <span className='polldate'> {p[1].date.slice(0, 10)}</span>
