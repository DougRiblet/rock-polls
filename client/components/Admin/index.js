import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { grabMyPolls } from '../../actions/actions';
import adminPresent from './admin-presentational';

const mapStateToProps = state => ({
  user_id: state.auth.user_id,
  username: state.auth.username,
  myPolls: state.mypolls,
});

const mapDispatchToProps = dispatch => bindActionCreators({ grabMyPolls }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(adminPresent);

