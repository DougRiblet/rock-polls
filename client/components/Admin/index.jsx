import React from 'react';

const Admin = () => (
  <div>
    ## ADMIN PAGE ##
  </div>
);

export default Admin;
