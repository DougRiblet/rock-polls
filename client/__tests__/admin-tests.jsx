import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Admin from '../components/Admin/admin-presentational';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('Admin', () => {
  test('Snapshot', () => {
    const tree = renderer.create(<Admin
      grabMyPolls={() => jest.fn()}
      myPolls={{}}
      user_id=''
      username=''
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Admin
      grabMyPolls={() => jest.fn()}
      myPolls={{}}
      user_id=''
      username=''
    />, div);
  });
});
