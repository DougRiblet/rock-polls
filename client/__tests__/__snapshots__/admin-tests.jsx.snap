// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Snapshot 1`] = `
<div
  id="show-my-polls"
>
  <div
    id="admin-head"
  >
    <h2>
      Polls Created by 
      
    </h2>
  </div>
  <div
    id="admin-list"
  >
    <p
      className="loading"
    >
      No Polls Found
    </p>
  </div>
  <div
    id="add-new-poll"
  >
    <Link
      to="/admin/create"
    >
      <button
        id="create-new-button"
      >
        Create New Poll
      </button>
    </Link>
  </div>
</div>
`;
